neqo-udp/src/lib.rs

```diff
@@ -20,7 +20,29 @@ use quinn_udp::{EcnCodepoint, RecvMeta, Transmit, UdpSocketState};
 /// Allows reading multiple datagrams in a single [`Socket::recv`] call.
 //
 // TODO: Experiment with different values across platforms.
-pub const RECV_BUF_SIZE: usize = u16::MAX as usize;
+const RECV_BUF_SIZE: usize = u16::MAX as usize;
+#[cfg(not(any(
+    target_os = "macos",
+    target_os = "ios",
+    target_os = "tvos",
+    target_os = "visionos"
+)))]
+const BATCH_SIZE: usize = 1;
+#[cfg(any(
+    target_os = "macos",
+    target_os = "ios",
+    target_os = "tvos",
+    target_os = "visionos"
+))]
+const BATCH_SIZE: usize = 32;
```

Comment:
Needs some commentary about why there is a split and what the benefits of the additional allocation are.  2MB is fine, but knowing why is important.

Reply:
Added rational for the platform specific values. Unfortunately we don't have good data, thus the concrete values are based on local testing only. Please take another look.